src/dune_rules/exe_rules.ml

```diff
@@ -182,6 +182,8 @@ let executables_rules
   let lib_config = ocaml.lib_config in
   let stdlib_dir = lib_config.stdlib_dir in
   let* requires_compile = Compilation_context.requires_compile cctx in
+  let* requires_link = Compilation_context.requires_link cctx in
+  let requires_link = Resolve.map ~f:(List.map ~f:(fun lib -> lib, true)) requires_link in
```

Comment:
What is the reason behind this change? I don't think this is not enough to fix Merlin. Better to do nothing than a change that is not really a fix.

Reply:
The only reason I had in my head is to keep `merlin.ml` coherent with the new bool. But you are right, it is not intended as a fix. I will remove it.